src/components/PostDetails.tsx

```diff
@@ -104,23 +59,23 @@ export const PostDetails: React.FC<Props> = ({ post }) => {
       </div>
 
       <div className="block">
-        {!loaded && (
+        {loaded && (
           <Loader />
         )}
 
-        {loaded && hasError && (
+        {!loaded && hasError && (
           <div className="notification is-danger" data-cy="CommentsError">
             Something went wrong
           </div>
         )}
 
-        {loaded && !hasError && comments.length === 0 && (
+        {!loaded && !hasError && comments.length === 0 && (
```

Comment:
[Pls check this](https://developer.mozilla.org/en-US/docs/Glossary/Falsy)
```suggestion
        {!loaded && !hasError && !comments.length && (
```